modules/internal/ChapelArray.chpl

```diff
@@ -1788,9 +1808,41 @@ module ChapelArray {
       _value.dsiRequestCapacity(i);
     }
 
-    /* Return the number of indices in this domain */
-    proc size return _value.dsiNumIndices;
-    /* Return the lowest index in this domain */
+    /* Return the number of indices in this domain.  For domains whose
+       ``intIdxType != int``, please refer to the note in
+       :proc:`ChapelRange.range.size` about changes to this routine's return type.
+    */
+    proc size {
```

Comment:
I wonder if `size` and `sizeas` should be inlined. May be a bad idea for the overflow check in sizeas, but we could throw that into a helper function

Reply:
I'm going to argue "no" for this PR, based on the following:
* it wasn't prior to these changes when it was arguably simpler
* it's not clear to me that it will appear in time-critical code
* for it to be valuable, I suspect we'd also need to inline the dsiNum*() routines as well?
* I've become somewhat wary of premature optimization through inlining

If someone else did a study that found that inlining these was beneficial to compile-time or execution-time, I definitely wouldn't fight it, but it doesn't feel like it fits this PR's theme or is something we should take on without more evidence.

Reply:
Sounds good, and I agree with not wanting to inline prematurely. The case I was wondering about is whether we call .size when indexing into arrays and if there's any benefit there. Or really I was thinking about it for block dist dsiAccess, but regardless I agree with saving this for a different day and checking that there would be a perf benefit first.